nixos/modules/services/misc/mesos-slave.nix

```diff
@@ -43,6 +51,12 @@ in {
         type = types.bool;
       };
 
+      withDocker = mkOption {
```

Comment:
@rushmorem @wmertens  I've updated the PR. Didn't use `mkDefault` as I couldn't figure out how to use that properly in this context. However, just setting `default = config.virtualisation.docker.enable` seems to work. With that I think the state of this PR is good.